graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -69,17 +69,99 @@ def _get_last_edge_direction_and_name_to_location(location):
+):
+    """Probe for the count of edges that connect parent_location and child_location vertices."""
+    _, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    outbound_vertex_name, inbound_vertex_name = _get_outbound_and_inbound_names_from_locations(
+        query_metadata, child_location, parent_location
+    )
+    probe_result = statistics.get_vertex_edge_vertex_count(
+        outbound_vertex_name, edge_name, inbound_vertex_name
+    )
+    return probe_result
+
+
+def _estimate_vertex_edge_vertex_count_using_class_count(
+    schema_graph, statistics, query_metadata, child_location, parent_location
+):
+    """Estimate the count of edges that connect parent_location and child_location vertices.
+
+    Given a parent location of class A and a child location of class B, this function estimates the
+    number of AB edges using class counts. If A and B are subclasses of the edge's endpoint classes
+    (which we'll name C and D respectively), we only have statistics for CD edges. So estimates for
+    the number of AB edges will be made using the assumption that CD edges are distributed
+    independently of whether or not the vertex of class C is also of class A and likewise for D and
+    B. In the general case, we estimate the statistic as
+    (# of AB edges) = (# of CD edges) * (# of A vertices) / (# of C vertices) *
+                                        (# of B vertices) / (# of D vertices).
+
+    Args:
+        schema_graph: SchemaGraph object
+        statistics: Statistics object
+        query_metadata: QueryMetadataTable object
+        child_location: BaseLocation object
+        parent_location: BaseLocation object
+
+    Returns:
+        float, estimate for (# of AB edges)
+    """
+    _, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    edge_counts = statistics.get_class_count(edge_name)
+
+    parent_name_from_location = query_metadata.get_location_info(parent_location).type.name
+    child_name_from_location = query_metadata.get_location_info(child_location).type.name
+    parent_base_classname, child_base_classname = _get_parent_and_child_base_classnames_from_edge(
+        schema_graph, child_location
+    )
+    # Scale edge_counts if child_location's type is a subclass of the edge's endpoint type.
+    if child_name_from_location != child_base_classname:
+        edge_counts *= (
+            float(statistics.get_class_count(child_name_from_location)) /
+            statistics.get_class_count(child_base_classname)
+        )
+    # Scale edge_counts if parent_location's type is a subclass of the edge's endpoint type.
+    if parent_name_from_location != parent_base_classname:
+        edge_counts *= (
+            float(statistics.get_class_count(parent_name_from_location)) /
+            statistics.get_class_count(parent_base_classname)
+        )
+
+    return edge_counts
 
 
 def _estimate_children_per_parent(
```

Comment:
This function should be named something more akin to `_estimate_parent_child_edges` because that is exactly what it is doing. Estimating the number of children vertices per parent vertex is slightly different.

Reply:
Refactoring this function is not the goal of this PR. 

If you'd like, create a PR for this change. I think it's going to be worthwhile to have someone else interacting with the cost_estimator, as I won't be here forever.

Reply:
Because of the time CI integration tests take, it is not practical to be making PRs for tiny nits like this one. I don't see how submitting a PR for this nit would make me any more experienced in dealing with the query cost estimator. Even though this is not the primary goal of this pr, I don't see the harm of making this change and it's related to this PR because you are changing the function's docstring.

Reply:
Since refactoring is now a part of this PR, I've changed the function name to `_estimate_edges_to_children_per_parent`.